src/components/projectSelect/index.tsx

```diff
@@ -62,18 +84,24 @@ export const ProjectSelect: React.FC<ProjectSelectProps> = ({
   return (
     <SearchableDropdown
       label="Project"
-      value={selectedProject?.displayName || selectedProject?.identifier}
+      value={
+        selectedProject?.displayName ||
+        selectedProject?.identifier ||
+        selectedProjectIdentifier
+      }
       options={allProjects}
       onChange={(projectIdentifier: any) => {
         Cookies.set(CURRENT_PROJECT, projectIdentifier, { expires: 365 });
-        history.push(getCommitsRoute(projectIdentifier));
+        history.push(getRoute(projectIdentifier));
       }}
       optionRenderer={(projectGroup, onClick) => (
         <ProjectOptionGroup
-          key={projectGroup.name}
+          key={projectGroup.name + projectGroup.projects.length}
```

Comment:
This won't prevent duplicate keys if this was your intent with this change. Since you can have 2 project groups with the same length.

Reply:
true. removing.